index.js

```diff
@@ -84,23 +84,24 @@ exports.lintText = (str, opts) => {
 	return processReport(report, opts);
 };
 
-exports.lintFiles = (patterns, opts) => {
+exports.lintFiles = (inputPatterns, opts) => {
 	opts = optionsManager.preprocess(opts);
-	patterns = patterns.length === 0 ? ['**/*'] : arrify(patterns);
 
+	const defaultPattern = `**/*.{${opts.extensions.join(',')}}`;
+	const patterns = inputPatterns.length === 0 ? [defaultPattern] : arrify(inputPatterns);
```

Comment:
Using two very similar variables/constants (`inputPatterns` and `patterns`)  within the same scope makes it harder to read and maintain IMO. As far as I can see, below your line 91, `inputPatterns.length` is the only important information, therefore we could use a constant like `patternsAreEmpty` instead:

```js
exports.lintFiles = (patterns, opts) => {
	//…

	const patternsAreEmpty = patterns.length === 0;
	patterns = patternsAreEmpty ? [defaultPattern] : arrify(patterns);

	//…

	return globby(patterns, …

		//…
		if (!patternsAreEmpty) {
```